### PR TITLE
Jsonstructured return changed information

### PR DESCRIPTION
**Pull request by manos-transifex**
Problem and/or solution
-----------------------
Support the display of changed stringset values in compilation.
- context (e.g. "something")
- developer_comments (e.g. "some_comments")
- character_limit (e.g. 149)

1) When we parse the source into a template and a stringset we are taking some extra information for each string

2) In the next step,  we compile a template and we provide a stringset and the template.
> but there is a possibility that one of the above 3 values has changed in the provided stringsset.

The cases that are handled
- if a value exists in the template -> then change the value with the provided.
- if an element does not exist in the template but a value exists in the string then append it at the end of the relevant object
- 
How to test
-----------

Reviewer checklist
------------------

Code:
* [ ] Change is covered by unit-tests
* [ ] Code is well documented, well styled and is following [best practices](https://tem.transifex.com)
* [ ] Performance issues have been taken under consideration
* [ ] Errors and other edge-cases are handled properly

PR:
* [ ] Problem and/or solution are well-explained
* [ ] Commits have been squashed so that each one has a clear purpose
* [ ] Commits have a proper commit message [according to TEM](https://tem.transifex.com/github-guide.html#working-on-a-feature)

### CODE REVIEW

File: openformats/formats/json.py [hunk 1]
```diff
@@ -488,6 +488,101 @@ class StructuredJsonHandler(JsonHandler):
...
+
+    def _compile_recursively(self, current_part):
+        if isinstance(current_part, DumbJson):
+            if current_part.type == list:
+                for value, value_position in current_part:
+                    self.transcriber.mark_section_start()
+                    self._compile_recursively(value)
+                    self.transcriber.mark_section_end()
+            if current_part.type == dict:
+                (value, pos) = current_part.find_children(self.STRING_KEY)[0]
```
Comment (igavriil) [inline openformats/formats/json.py:509]: minor: `pos` is unused so maybe instead of `(value, pos)` -> `(value, _)`

File: openformats/formats/json.py [hunk 2]
```diff
@@ -488,6 +488,101 @@ class StructuredJsonHandler(JsonHandler):
...
+                if value is None:
+                    for key, key_position, value, value_position in current_part:
+                        self.transcriber.copy_until(key_position - 1)
+                        self.transcriber.mark_section_start()
```
Comment (igavriil) [inline openformats/formats/json.py:513]: `self.transcriber.mark_section_start()`
and
`self.transcriber.mark_section_end()`
can be removed since no sections are remove on compilation

File: openformats/formats/json.py [hunk 3]
```diff
@@ -488,6 +488,101 @@ class StructuredJsonHandler(JsonHandler):
...
+                        self.transcriber.copy_until(value_position)
+                        if key == self.CONTEXT_KEY and translation:
+                            if translation.context is not None:
+                                self.transcriber.add(translation.context)
+                            else:
+                                self.transcriber.add(u"null")
+                            self.transcriber.skip(len(str(value)))
+                            self.transcriber.copy_until(value_position + len(str(value)) + 1)
```
Comment (igavriil) [inline openformats/formats/json.py:537]: (Suggestion) Maybe add a function to reuse:
```
def _compile_value(self, value, template_value, value_position):
    if value is not None:
        self.transcriber.add(value)
    else:
        self.transcriber.add(u"null")
    self.transcriber.skip(len(template_value))
    self.transcriber.copy_until(value_position + len(template_value) + 1)
```
Then you can do:

```
if key == self.CONTEXT_KEY and translation:
    context = translation.context
    self._compile_value(context, value, value_position)
    context_added = True
elif key == self.DEVELOPER_COMMENT_KEY and translation:
    developer_comment = translation.developer_comment
    self._compile_value(developer_comment, value, value_position)
    developer_comments_added = True
elif key == self.CHARACTER_LIMIT_KEY and translation:
    character_limit = u"{}".format(translation.character_limit)
    value = u"{}".format(value)
    self._compile_value(character_limit, value, value_position)
    character_limit_added = True
elif key == self.STRING_KEY and translation:
    key_value_separator = current_part.source[key_position+len(key):value_position-1]
    if translation.pluralized:
        string_replacement = ICUCompiler().serialize_strings(translation.string, delimiter=' ')
    else:
        string_replacement = translation.string
    self._compile_value(string_replacement, value, value_position)
```

**Review by igavriil — APPROVED** [review]
Great and detailed work!
Kudos :tada:

**Review by igavriil — APPROVED** [review]
